### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by mkangia**
Found these while trying to learn about restores

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 1]
```diff
@@ -698,7 +698,7 @@ def set_cached_payload_if_necessary(self, fileobj, duration, is_async):
         return None
 
     def delete_cached_payload_if_necessary(self):
-        if self.overwrite_cache and self.restore_payload_path_cache.get_value():
+        if self.overwrite_cache and self.restore_payload_path_cache.exists():
```
Comment (mkangia) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: Made an assumption there that the value can never be a falsy value so it was fine to just check for the presence of the key.

File: corehq/ex-submodules/casexml/apps/phone/restore_caching.py [hunk 2]
```diff
@@ -16,6 +16,10 @@ class _CacheAccessor(object):
     timeout = None
     debug_info = None
 
+    def exists(self):
+        logger.debug('if exists {}'.format(self.debug_info))
+        return get_redis_default_cache().has_key(self.cache_key)
```
[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: Fine to ignore this?

**Review by snopoke — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

File: corehq/ex-submodules/casexml/apps/phone/restore_caching.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: The non-deprecated way is preferred: `self.cache_key in get_redis_default_cache()`

[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: Thanks for taking a look at that @millerdev .
I was expecting `has_key` to be a method call on a cache object and hence could not understand the suggestion there. Is there some magic happening here?

[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: They've improved the redis object make use of the `in` operator, which yields more natural phrasing. Under the hood, `in` results in something like `get_redis_default_cache().__contains__(self.cache_key)`. Note, it is usually not necessary to call "dunder" methods (ones beginning and ending with `__`) explicitly as they are normally an implementation detail behind a nicer syntactical structure.

[the remark this answers is not in the document]
Comment (mkangia) [reply corehq/ex-submodules/casexml/apps/phone/restore_caching.py]: thank you for the details @millerdev .. much appreciated :)
will make the change

**Review by millerdev — APPROVED** [review]

**Review by snopoke — APPROVED** [review]

**Pull request merged**